tools/clang/lib/SPIRV/EmitVisitor.cpp

```diff
@@ -534,9 +592,9 @@ bool EmitVisitor::visit(SpirvSource *inst) {
       debugMainFileId = fileId;
   }
 
-  if (dumpedFiles.count(fileId) != 0)
+  if (emittedSource[fileId] != 0)
     return true;
-  dumpedFiles.insert(fileId);
+  emittedSource[fileId] = 1;
```

Comment:
Make sense. I was confused that another `fileId` can have `1` for its `DebugSource` and this can cause a conflict. Could you please add an explanation to describe sth like "Note that it is fine that another `DebugSource` has `1` as its result id, because in the case of `OpSource` it does not have a result id and `emittedSource` is only used to check whether the `OpSource` was emitted or not."?

In addition, using a constant uint variable instead of `1` would be better e.g., `static const uint kEmittedSourceForOpSource = 1; // Since OpSource does not have a result id, this is used to mark it was emitted`.

If you want to delay the refactoring for this update because of the conflicts with commits on the top of this change, let me know. I will just approve and submit it.